### DISCUSSION TITLE
[Clarification] : Type of trace file of extension .log

### DISCUSSION THREAD

**Discussion by anubhavjana** [ChampSim/ChampSim#405] [Q&A]
Hello, wanted to know if we can run trace files with .log extension i.e. pass .log file to champsim executable instead of .xz?

**Comment by wilsonwang881** [comment]
Please refer to `src/tracereader.cc` line 38 to line 40. Looks like you need to have the files end with *gz*, *xz* or *bz2*.

You can change *log* extension to any of the one mentioned above but the file should be compressed in one of the above 3 formats.

Hope that helps.

**Comment by anubhavjana** [comment]
Thanks @wilsonwang881 for your response. Had a look at the source file. Will try that out.

**Comment by ngober** [comment]
The trace file does not need to be compressed. If you pass an uncompressed file, the simulator will read it just fine. We usually compress them because practically useful traces can often be quite large.

Note that the compression detection is very rudimentary: it makes an assumption based on the file name. So, if you are passing a zlib-compressed trace named `picture_of_cat.jpg`, the simulator will give you, well, unhelpful results. One of these days someone should get around to better compression detection (:point_right: #382).

**Reply by anubhavjana** [reply]
Thanks , so if i pass any other extension apart from these 3 , it will go the last "else" and  read it just fine right? @ngober 

if (is_gzip_compressed)
    return champsim::tracereader{R<T, champsim::inf_istream<champsim::decomp_tags::gzip_tag_t<>>>(cpu, fname)};
  else if (is_lzma_compressed)
    return champsim::tracereader{R<T, champsim::inf_istream<champsim::decomp_tags::lzma_tag_t<>>>(cpu, fname)};
  else if (is_bzip2_compressed)
    return champsim::tracereader{R<T, champsim::inf_istream<champsim::decomp_tags::bzip2_tag_t>>(cpu, fname)};
  else
    **return champsim::tracereader{R<T, std::ifstream>(cpu, fname)};**

**Reply by ngober** [reply]
That is correct.
